ivy/functional/frontends/paddle/nn/functional/vision.py

```diff
@@ -170,3 +170,105 @@ def pixel_shuffle(x, upscale_factor, data_format="NCHW"):
     return ivy.reshape(
         ivy.permute_dims(input_reshaped, (0, 1, 4, 2, 5, 3)), (b, oh, ow, oc)
     )
+@to_ivy_arrays_and_back
+def grid_sample(x, grid, mode='bilinear', padding_mode='zeros', align_corners=True):
+
+    _modes = ['bilinear', 'nearest']
+    _padding_modes = ['zeros', 'reflection', 'border']
+    if mode not in _modes:
+        raise ValueError(
+            "The mode of grid sample function should be in {}, but got: {}".format(
+                _modes, mode
+            )
+        )
+    if padding_mode not in _padding_modes:
+        raise ValueError(
+            "The padding mode of grid sample function should be in {}, but got: {}".format(
+                _padding_modes, padding_mode
+            )
+        )
+
+    if not isinstance(align_corners, bool):
+        raise ValueError(
+            "The align corners should be bool, but got: {}".format(
+                align_corners
+            )
+        )
+    N, C, H, W = x.shape
+    if len(grid.shape) == 4:
+        _, grid_H, grid_W, _ = grid.shape
+    elif len(grid.shape) == 5:
+        _, _, grid_D, grid_H, grid_W = grid.shape
+    else:
+        raise ValueError("Invalid grid shape")
+
+    grid_x = grid[:, :, :, 0]
+    grid_y = grid[:, :, :, 1]
+
+    if align_corners:
+        grid_x = 0.5 * (grid_x + 1) * (W - 1)
+        grid_y = 0.5 * (grid_y + 1) * (H - 1)
+    else:
+        grid_x = 0.5 * grid_x * (W - 1)
+        grid_y = 0.5 * grid_y * (H - 1)
+
+    y_t = ivy.zeros((N, C, grid_H, grid_W), dtype=x.dtype)
+
+    for n in range(N):
+        for c in range(C):
+            for i in range(grid_H):
+                for j in range(grid_W):
```

Comment:
It should be possible to do this without using nested loops, which would be far more efficient. Maybe have a look at this implementation of grid_sample which was recently implemented in the torch frontend for inspiration? https://github.com/unifyai/ivy/pull/22539

Reply:
Thank you @Sam-Armstrong for the resources provided have been able to remove the forloops now lert me proceed to the tests